provider_shopper/lib/models/cart.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:flutter/foundation.dart';
+import 'package:provider_shopper/models/catalog.dart';
+
+class CartModel extends ChangeNotifier {
+  /// The current catalog. Used to construct items from numeric ids.
+  final CatalogModel _catalog;
+
+  /// Internal, private state of the cart. Stores the ids of each item.
+  final List<int> _itemIds;
+
+  CartModel(this._catalog, CartModel previous)
```

Comment:
nit: Throw a comment on here to explain when this would be used (also, it's the only method in here with no comment).